### PR TITLE
Add examples for widgets methods

### PR DESCRIPTION
**Pull request by abhiramtilakiiit**
### Overview

Add minimal but informative, examples for different functions in `widgets.py` as referenced in #1629.

Also fixed an existing mismatch in examples.

### Details

The examples were added to the following functions in `widgets.py` file:

- `add_mesh_clip_box`
- `add_mesh_clip_plane`
- `add_mesh_clip_isovalue`
- `add_mesh_slice`
- `add_mesh_slice_orthogonal`

The example in function `add_box_widget`, doesn't explicitly use the function, instead uses the `add_mesh_clip_box`, but in my opinion it deserves its own example. So, I have moved the example under `add_box_widget` into `add_mesh_clip_box`.

### CODE REVIEW

File: pyvista/plotting/widgets.py [hunk 1]
```diff
@@ -153,18 +153,6 @@ def add_box_widget(
...
-        Shows an interactive clip box.
-
-        >>> import pyvista as pv
-        >>> mesh = pv.ParametricConicSpiral()
-        >>> pl = pv.Plotter()
-        >>> _ = pl.add_mesh_clip_box(mesh, color='white')
-        >>> pl.show()
-
-        For a full example see :ref:`box_widget_example`.
-
```
Comment (tkoyama010) [inline pyvista/plotting/widgets.py:167]: Thank you for moving the example to another correct place. Could you add the revised example to the deleted section?

**Review by tkoyama010 — CHANGES_REQUESTED** [review]

File: pyvista/plotting/widgets.py (hunk 1 above)
Comment (abhiramtilakiiit) [reply pyvista/plotting/widgets.py:167]: I have more examples for `add_box_widget` and `add_line_widget`, but I'm not sure, if I am
doing it the right way. 
I also want to add `add_plane widget `, but I'm unable to come up with a better example than one given in the [examples](https://docs.pyvista.org/version/stable/examples/03-widgets/plane-widget.html) with carotid.

Comment (tkoyama010) [reply pyvista/plotting/widgets.py:167]: @abhiramtilakiiit Thanks for comment.
Please commit any examples you are　considering. We can review and discuss them together. If you don't have any that you can commit, I will commit a draft and you can mentions me.

Comment (abhiramtilakiiit) [reply pyvista/plotting/widgets.py]: commit [f166a74](https://github.com/pyvista/pyvista/pull/5036/commits/f166a749b4866565a87289c79a9a1d735ed612c2) resolves this.

**Comment by tkoyama010** [comment]
.pre-commit.ci autofix

File: pyvista/plotting/widgets.py [hunk 2]
```diff
@@ -437,6 +467,33 @@ def add_plane_widget(
         vtk.vtkImplicitPlaneWidget or vtk.vtkPlaneWidget
             Plane widget.
 
+        Examples
+        --------
+        Shows an interactive plane moving along x-axis in random-hill example, which is used to mark the max altitude
+        at a particular distance x.
```
Comment (tkoyama010) [inline pyvista/plotting/widgets.py:473]:
```suggestion
        Shows an interactive plane moving along the x-axis in the random-hill example, which is used to mark the max altitude
        at a particular distance x.
```

**Review by tkoyama010 — APPROVED** [review]
LGTM. Thanks for your awesome work!
image
image
image
image
image
image
image

**Pull request merged**